diff --git a/update-client-hub/modules/atomic-queue/source/critical.c b/update-client-hub/modules/atomic-queue/source/critical.c
index 5d3c5e2..b540cf5 100644
--- a/update-client-hub/modules/atomic-queue/source/critical.c
+++ b/update-client-hub/modules/atomic-queue/source/critical.c
@@ -18,7 +18,7 @@
 
 // This critical section implementation is generic for mbed OS targets,
 // except Nordic ones
-#if defined(TARGET_LIKE_MBED) && !defined(TARGET_NORDIC)
+#if defined(TARGET_LIKE_MBED) && !defined(TARGET_NRF51)
 
 #include <stdint.h>                  // uint32_t, UINT32_MAX
 #include <stddef.h>                  // NULL
